cg/meta/rsync/rsync_api.py

```diff
@@ -167,62 +189,25 @@ def create_log_dir(self, dry_run: bool) -> None:
-    ) -> list[str]:
-        """Returns a list of all the folder names depending if sample and/or case data is to be
-        transferred."""
-        if not sample_files_present and not case_files_present:
-            raise CgError(
-                "Since neither case or sample files are present, no files will be transferred"
-            )
-        folder_list: list[str] = []
-        if sample_files_present:
-            folder_list.extend(
-                [sample.name for sample in self.status_db.get_samples_by_case_id(case_id=case_id)]
-            )
-        if case_files_present:
-            folder_list.append(self.status_db.get_case_by_internal_id(internal_id=case_id).name)
-        return folder_list
-
-    def slurm_rsync_single_case(
-        self,
-        case: Case,
-        dry_run: bool,
-        sample_files_present: bool = False,
-        case_files_present: bool = False,
-    ) -> tuple[bool, int]:
+    def run_rsync_for_case(self, case: Case, dry_run: bool, folders_to_deliver: set[Path]) -> int:
         """Runs rsync of a single case to the delivery server, parameters depend on delivery type."""
-        case_id: str = case.internal_id
-
-        ticket: str = self.status_db.get_latest_ticket_from_case(case_id=case_id)
+        ticket: str = self.status_db.get_latest_ticket_from_case(case_id=case.internal_id)
         source_and_destination_paths: dict[str, Path] = self.get_source_and_destination_paths(
             ticket=ticket, customer_internal_id=case.customer.internal_id
         )
-        self.set_log_dir(folder_prefix=case_id)
+        self.set_log_dir(folder_prefix=case.internal_id)
         self.create_log_dir(dry_run=dry_run)
-
-        folders: list[str] = self.get_folders_to_deliver(
-            case_id=case_id,
-            sample_files_present=sample_files_present,
-            case_files_present=case_files_present,
-        )
-        existing_folders: list[str] = [
-            folder
-            for folder in folders
-            if Path(source_and_destination_paths["delivery_source_path"], folder).exists()
-        ]
-        commands: str = RsyncAPI.concatenate_rsync_commands(
-            folder_list=existing_folders,
+        commands: str = self.concatenate_rsync_commands(
+            folder_list=folders_to_deliver,
             source_and_destination_paths=source_and_destination_paths,
             ticket=ticket,
+            case=case,
         )
-        is_complete_delivery: bool = folders == existing_folders
-        return is_complete_delivery, self.sbatch_rsync_commands(
-            commands=commands, job_prefix=case_id, dry_run=dry_run
+        return self.sbatch_rsync_commands(
+            commands=commands, job_prefix=case.internal_id, dry_run=dry_run
         )
 
-    def run_rsync_on_slurm(self, ticket: str, dry_run: bool) -> int:
+    def run_rsync_for_ticket(self, ticket: str, dry_run: bool) -> int:
         """Runs rsync of a whole ticket folder to the delivery server."""
```

Comment:
Most likely needs an update.

Reply:
Only the name changed for this function the docstring is correct.